code/modules/items/specials/ItemSpecials.dm

```diff
@@ -1142,6 +1142,8 @@ ABSTRACT_TYPE(/datum/item_special/spark)
 	pixelaction(atom/target, params, mob/user, reach)
 		if(!isturf(target.loc) && !isturf(target)) return
 		if(!usable(user)) return
+		if(istype(user:gloves, /obj/item/clothing/gloves/bladed))
+			SEND_SIGNAL(user, COMSIG_MOB_CLOAKING_DEVICE_DEACTIVATE)
```

Comment:
is this needed?

Reply:
it is, else the special attack does not deactivate the cloaker